app/models/campaign_page.rb

```diff
@@ -7,16 +7,25 @@ class CampaignPage < ActiveRecord::Base
   has_many :campaign_pages_widgets, inverse_of: :campaign_page
   has_many :campaign_pages_tags
   has_many :tags, through: :campaign_pages_tags, source: :tag
+  has_many :widgets, dependent: :destroy
```

Comment:
A campaign page now has both many widgets and many campaign_pages_widgets - what's the difference? I see we removed validates_associated for campaign_pages_widgets.